### PR TITLE
 Add no binding this & argument testcase

### PR DESCRIPTION
**Pull request by b-barry**

### CODE REVIEW

File: test/arrow-function.spec.js [hunk 1]
```diff
@@ -1,13 +1,59 @@
...
+      function foo() {
+        const f = function(i) { arguments[0] + i };
+        return f(2);
+      }
+
+      actual = foo(1);
+
+      expect(actual).equal(__);
+
+      function bar() {
```
Comment (davyengone) [inline test/arrow-function.spec.js]: where `bar`is used?

File: test/arrow-function.spec.js [hunk 2]
```diff
@@ -1,13 +1,59 @@
...
+  
+  describe('No binding of this & arguments ', () => {
+    it("without its own 'this' context", () => {
+
+      // TODO Modify the growUp function above to satisfy all the assertions below. Using '.bind' is forbidden
+      function Person() {
+        this.age = 10;
+
+        setTimeout(function growUp() {
+          this.age++;
```
Comment (davyengone) [inline test/arrow-function.spec.js:94]: this will never update the value `this.age`

File: test/arrow-function.spec.js [hunk 3]
```diff
@@ -1,13 +1,59 @@
...
 describe('Arrow function', () => {
+  
+  describe('No binding of this & arguments ', () => {
+    it("without its own 'this' context", () => {
+
+      // TODO Modify the growUp function above to satisfy all the assertions below. Using '.bind' is forbidden
+      function Person() {
+        this.age = 10;
+
+        setTimeout(function growUp() {
```
Comment (davyengone) [inline test/arrow-function.spec.js]: when you pass callback, just use anonymous `function`

File: test/arrow-function.spec.js [hunk 4]
```diff
@@ -1,13 +1,59 @@
...
+          this.age++;
+        }, 1000);
+      }
+
+      const person = new Person();
+
+      expect(person.age).equal(11);
```
Comment (davyengone) [inline test/arrow-function.spec.js]: This will never be 11 and you should use `const actual = person.age`

**Review by davyengone — CHANGES_REQUESTED** [review]
could you deal with my comments, please.

**Comment by b-barry** [comment]
For the growUp function, I corrected the TODO sentence to explain the goal.
Added the actual statement

File: test/arrow-function.spec.js [hunk 5]
```diff
@@ -1,13 +1,62 @@
...
+    it("without its own 'this' context", (done) => {
+
+      // TODO Modify the growUp function below to satisfy all the assertions below. Using '.bind' is forbidden
+      function Person() {
+        this.age = 10;
+
+        setTimeout(function growUp() {
+          this.age++;
+          expect(actual).equal(11);
+          done();
```
Comment (hwaterke) [inline test/arrow-function.spec.js]: Maybe worth adding a comment to explain what done() does here

File: test/arrow-function.spec.js [hunk 6]
```diff
@@ -1,13 +1,62 @@
...
+          
+        }, 2000);
+      }
+
+      const person = new Person();
+      const actual = person.age;
+
+    });
+    it("without its own 'argument' object ", () => {
```
Comment (hwaterke) [inline test/arrow-function.spec.js]: For me, this is should be one describe() and 3 it()

**Review by hwaterke — CHANGES_REQUESTED** [review]

**Review by hwaterke — APPROVED** [review]

**Pull request merged**